x-pack/legacy/plugins/siem/server/lib/framework/kibana_framework_adapter.ts

```diff
@@ -149,16 +152,40 @@ export class KibanaBackendFrameworkAdapter implements FrameworkAdapter {
             request
           );
 
-          return h.response(graphiqlString).type('text/html');
+          return response.ok({
+            body: graphiqlString,
+            headers: {
+              'content-type': 'application/json',
+            },
+          });
+        }
+      );
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private handleError(error: any, response: KibanaResponseFactory) {
+    if (error.name !== 'HttpQueryError') {
+      return response.internalError({
+        body: error.message,
+        headers: {
+          'content-type': 'application/json',
         },
-        method: 'GET',
-        path: `${routePath}/graphiql`,
       });
     }
+
+    return response.customError({
```

Comment:
Is there currently a way to log this error info? A generic "Internal Server Error" message (or none at all if it's a 4xx) with a trace originating from the [Boom instantiation](https://github.com/elastic/kibana/blob/master/src/core/server/http/router/response_adapter.ts#L118-L120) is logged here, which isn't as helpful.

Reply:
You can use the NP logger to construct your own error logs.